test/src/responder.test.ts

```diff
@@ -0,0 +1,627 @@
+import * as chai from "chai";
```

Comment:
I think these tests are good, but as you've pointed they're more like integration tests. I think for that reason we should keep them as they are. However i think the reason they're like this is because responder code is too procedural, and not OOP enough. It means that the unit under test for the responder is very large making it hard to ensure that the test coverage is high. In the responder file Ive suggested some ways in which we can break up the responder from 1 class into 5. This will allow you to test each of these classes separately - making for smaller, simpler tests. I would suggest keeping these tests as they are and consider them intergration tests, but add in unit tests for all the smaller classes we'll write. I would also move the MultiResponder tests to a new branch, along with the multiresponder so we can work on it later.